ls1intum/Apollon
# [Feature] Add grading instruction link for uml element feedback

### Checklist

### Motivation and Context



When the grading instruction is associated with feedback, there is no way to see this relation in UI. Also, the assessor can not remove this association directly in assessment view. To achieve this issue, simply `link icon` is added to feedback block. When the assessor dropped the structured grading instruction into feedback, the link is appeared. When the user hovers over it, the small popup displays the associated grading instruction description. After one click, `link icon` changes into `trash icon` and the popup warns user. After second click, the association between structured grading instruction and feedback is removed.

PS: Currently, I can not send the tooltip messages from Artemis. Also, when you change the language in Artemis, the strings in Apollon editor keeps same. So I added them as plain text for now. The translation issue can be managed later on.

### Description


### Steps for Testing


Log in to Artemis
1.Navigate to Course Administration
2.Create/Find a new exercise.
3.Add an example solution for the exercise / participate this exercise as a student.
   a. If you add an example submission, then create new assessment for this submission and use one of the structured grading instruction as a feedback.
   b. If you participated this exercise as student, then login as tutor/instructor and assess this submission. During the assessment, use one of the structured grading instructions as a feedback. (drag the instruction and drop into the class/attribute/method in uml)
4.You should see the link icon like in the following screenshots.
5.Mouse over it, check the popup.
6.Click one time, see the delete icon and popup. Click again. link should be disappeared.

Additional Steps
1. Confirm that the link icon is not shown in modeling editor as a student after your submission is graded by tutor.
a. assess your submission as a tutor
b. login as student, check your submission with its assessment.
2. Confirm that in the working mode of editor, link icon is not shown. (You can test this for Example Submission. When you change assessment mode, link icon should not be there.)

_Training Tutor mode_
3. Create example submission as Instructor. Add assessment using grading instructions. Save it. 
4. Login as Tutor. Go to Exercise Dashboard and start your training.
5. Check the assessment, you should see "link icon", but when you click you should not see "trash icon". tutor should not be able to remove the association between feedback and grading instruction. This editor mode is read-only.

### Screenshots



<img src="https://user-images.githubusercontent.com/23650630/123714801-4da07000-d877-11eb-9b2e-350f1e345b7c.png" width="500" />
<img src="https://user-images.githubusercontent.com/23650630/123714823-5b55f580-d877-11eb-80d4-1d4e9de361a7.png" width="500" />


###Test Coverage
assessment-dropInfo-tooltip.tsx - 100%
